service/src/main/java/org/mskcc/cmo/metadb/service/impl/MessageHandlingServiceImpl.java

```diff
@@ -210,49 +222,103 @@ public void run() {
                                     .buildNewResearchSampleFromMetadata(sampleMetadata.getIgoRequestId(),
                                             sampleMetadata);
-                            sampleService.saveSampleMetadata(sample);
-                            LOG.info("Publishing metadata history for new sample: "
+                            sampleService.saveMetadbSample(sample);
+                            LOG.info("Publishing metadata history for new research sample: "
                                     + sampleMetadata.getPrimaryId());
                             messagingGateway.publish(CMO_SAMPLE_UPDATE_TOPIC,
                                     mapper.writeValueAsString(sample.getSampleMetadataList()));
                         } else if (sampleService.sampleHasMetadataUpdates(
                                 existingSample.getLatestSampleMetadata(), sampleMetadata)) {
-                            LOG.info("Found updates for sample - persisting to database: "
+                            LOG.info("Found updates for research sample - persisting to database: "
                                     + sampleMetadata.getPrimaryId());
                             // persist sample level updates to database and publish
                             // sample metadata history to CMO_SAMPLE_METADATA_UPDATE
                             existingSample.updateSampleMetadata(sampleMetadata);
-                            sampleService.saveSampleMetadata(existingSample);
-                            LOG.info("Publishing sample-level metadata history for sample: "
+                            sampleService.saveMetadbSample(existingSample);
+                            LOG.info("Publishing sample-level metadata history for research sample: "
                                     + sampleMetadata.getPrimaryId());
                             messagingGateway.publish(CMO_SAMPLE_UPDATE_TOPIC,
                                     mapper.writeValueAsString(existingSample.getSampleMetadataList()));
                         } else {
-                            LOG.info("There are no updates to persist for sample: "
+                            LOG.info("There are no updates to persist for research sample: "
                                     + sampleMetadata.getPrimaryId());
                         }
                     }
-                    if (interrupted && sampleUpdateQueue.isEmpty()) {
+                    if (interrupted && researchSampleUpdateQueue.isEmpty()) {
+                        break;
+                    }
+                } catch (InterruptedException e) {
+                    interrupted = true;
+                } catch (Exception e) {
+                    LOG.error("Error during handling of research sample metadata update", e);
+                }
+            }
+            researchSampleUpdateHandlerShutdownLatch.countDown();
+        }
+    }
+
+
+    private class NewClinicalSampleMetadataHandler implements Runnable {
```

Comment:
rename suggestion: remove `New` from name?

Reply:
I introduced `new` to the name so later we could have a `updateHandler` and also to make it consistent with `newRequestHandler`